File: Back/src/main/java/com/adbd/Back/model/Partida.java
package com.adbd.Back.model;

import com.adbd.Back.enums.ResultadoEnum;
import lombok.*;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "partidas")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Partida {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;


    @ManyToOne(targetEntity = Jugador.class)
    @JoinColumn(name = "id_jugador", referencedColumnName = "id")
    private Jugador jugador;


    @ManyToMany
    @JoinColumn(name = "id_partida",referencedColumnName = "id")
    private List<Carta> cartas;

    @Enumerated(EnumType.STRING)
    private ResultadoEnum resultado;
}

File: Front/src/app/interfaces/Partida.ts
import { Carta } from './Carta';
import { Jugador } from './Jugador';

export interface Partida {
  id: number;
  jugador: Jugador;
  cartas: Carta[];
  resultado: ResultadoEnum;
  puntosJugador: number;
  puntosCroupier: number;
}
export enum ResultadoEnum {
  VICTORIA_JUGADOR = 'VICTORIA_JUGADOR',
  VICTORIA_CROUPIER = 'VICTORIA_CROUPIER',
  EMPATE = 'EMPATE',
}
